signinapp/model.py

```diff
@@ -51,14 +52,33 @@ class StampEvent():
     event: str
 
 
+class ShirtSizes(enum.Enum):
+    Small = 1
+    Medium = 2
+    Large = 3
+    X_Large = 4
+    XX_Large = 5
+    XXX_Large = 6
+
+    @classmethod
+    def get_size_names(cls):
+        return [(size.name, size.name.replace("_", "-")) for size in cls]
+
+
 class User(UserMixin, db.Model):
     __tablename__ = "users"
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String, nullable=False)
+    preferred_name = db.Column(db.String, nullable=True)
```

Comment:
Why is this `nullable=True` if the form has it as required?

Reply:
I've removed the `nullable=True` since it's the default (I think). I've also made this field optional as we will use the name instead if nothing is provided.